superset/superset/db_engine_specs.py

```diff
@@ -1416,6 +1421,74 @@ class SparkSQLEngineSpec(HiveEngineSpec):
 
     engine = 'sparksql'
 
+    @classmethod
+    def _dumps_operation_handle(cls, op_handle):
+        return {
+            'operation_id': {
+                'guid': op_handle.operationId.guid.decode('ISO-8859-1'),
+                'secret': op_handle.operationId.secret.decode('ISO-8859-1'),
+            },
+            'operation_type': op_handle.operationType,
+            'has_result_set': op_handle.hasResultSet,
+            'modified_row_count': op_handle.modifiedRowCount
+        }
+
+    @classmethod
+    def _loads_operation_handle(cls, op_handle):
+        from pyhive import hive
+
+        handle_id = hive.ttypes.THandleIdentifier(
+            guid=op_handle['operation_id']['guid'].encode('ISO-8859-1'),
+            secret=op_handle['operation_id']['secret'].encode('ISO-8859-1'),
+        )
+
+        return hive.ttypes.TOperationHandle(
+            operationId=handle_id,
+            operationType=op_handle['operation_type'],
+            hasResultSet=op_handle['has_result_set'],
+            modifiedRowCount=op_handle['modified_row_count']
+        )
+
+    @classmethod
+    def get_connection_id(cls, cursor):
+        """Returns connection id for a cursor
+
+        Just uses a hard-coded dummy id. This is done because queries
+        corresponding to a cursor without a connection id are interpreted as
+        non-cancellable.
+
+        A more suitable value should be the session id of the underlying
+        connection of the cursor, but that's not an integer, and here an
+        integer is required."""
+
+        return 1
```

Comment:
An alternative would be to change [this condition](https://github.com/src-d/sourced-ui/pull/267/files#diff-be6d5c3f88c01a7e34335ef082125245R326).